Sources/Plasma/PubUtilLib/plAudioCore/plSrtFileReader.cpp

```diff
@@ -0,0 +1,184 @@
+                            subtitleEndTimeMs += (atoi(matches[8].str().c_str()));
+                        }
+                    }
+                } else if (lnCounter % 4 == 2 && srtFileStream.ReadLn(line)) {
+                    if (std::regex_match(line.cbegin(), line.cend(), matches, speakerTextRegex)) {
+                        if (matches.size() < 5) {
+                            // add error message and ensure this subtitle line won't be added to the entries
+                            plStatusLog::AddLineSF("audio.log", plStatusLog::kRed, "   Subtitle text and/or speaker name '{}' are formatted incorrectly.", line);
+                            subtitleNumber = 0;
+                        } else {
+                            // matches[0] is the entire match, we don't do anything with it
+                            // matches[2] is the second group (the optional subtitle speaker with colon but no space at the end)
+                            speakerName = matches[2];
+
+                            // matches[5] is the fourth group (the subtitle text)
+                            subtitleText = matches[5];
+                        }
+                    }
+                } else if (lnCounter % 4 == 3 && subtitleNumber > 0 && subtitleStartTimeMs >= 0 && subtitleEndTimeMs >= 0 && !subtitleText.empty()) {
+                    // entry is complete, add to the queue and reset our temp variables
+                    if (!speakerName.empty())
+                        fEntries.emplace_back(subtitleNumber, subtitleStartTimeMs, subtitleEndTimeMs, std::move(subtitleText), std::move(speakerName));
+                    else
+                        fEntries.emplace_back(subtitleNumber, subtitleStartTimeMs, subtitleEndTimeMs, std::move(subtitleText));
+
+                    subtitleNumber = 0;
+                    subtitleStartTimeMs = 0;
+                    subtitleEndTimeMs = 0;
+                    subtitleText.clear();
+                    speakerName.clear();
+                }
+            }
+
+            if (subtitleNumber > 0 && subtitleStartTimeMs >= 0 && subtitleEndTimeMs >= 0 && !subtitleText.empty()) {
+                // enqueue the last subtitle from the file if we didn't have an extra blank line at the end
+                if (!speakerName.empty())
+                    fEntries.emplace_back(subtitleNumber, subtitleStartTimeMs, subtitleEndTimeMs, std::move(subtitleText), std::move(speakerName));
+                else
+                    fEntries.emplace_back(subtitleNumber, subtitleStartTimeMs, subtitleEndTimeMs, std::move(subtitleText));
+            }
+
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void plSrtFileReader::AdvanceToTime(uint32_t timeMs)
+{
+    if (fCurrentEntryIndex >= 0) {
+        while (fCurrentEntryIndex < fEntries.size() && fEntries.at(fCurrentEntryIndex).GetEndTimeMs() <= timeMs) {
+            fCurrentEntryIndex++;
+        }
+    }
```

Comment:
```suggestion
    while (fCurrentEntryIndex < fEntries.size() && fEntries.at(fCurrentEntryIndex).GetEndTimeMs() <= timeMs) {
        fCurrentEntryIndex++;
    }
```

`if (fCurrentEntryIndex >= 0)` will always be true because `fCurrentEntryIdx` is unsigned.